quipucords/scanner/vcenter/inspect.py

```diff
@@ -133,11 +133,10 @@ def parse_cluster_props(self, props, parents_dict):
             if prop.name == 'name':
                 facts['cluster.name'] = prop.val
             elif prop.name == 'parent':
-                parent = parents_dict[str(prop.val)]
-                while parent['type'] != 'vim.Datacenter':
-                    parent = parents_dict[parent['parent']]
-                facts['cluster.datacenter'] = parent['name']
-
+                parent = parents_dict.get(str(prop.val))
+                while parent.get('type') != 'vim.Datacenter':
```

Comment:
If `str(prop.val)` does not resolve to a key then parent will be None. Which means that calling `parent.get` will raise an error saying that None type does not have the get method.

Reply:
Great Catch! I will fix this asap.

Reply:
👍

Reply:
Added some conditions to catch this

Reply:
This still can raise an exception:

```python
In [1]: parents_dict = {}

In [2]: parent = parents_dict.get('some_key')

In [3]: parent is None
Out[3]: True

In [4]: parent.get('type')
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-4-716a922e2953> in <module>()
----> 1 parent.get('type')

AttributeError: 'NoneType' object has no attribute 'get'
```